mteb/abstasks/TaskMetadata.py

```diff
@@ -174,40 +174,60 @@ class TaskMetadata(BaseModel):
 
     name: str
     description: str
-    type: TASK_TYPE
-    modalities: list[Literal["text"]]
-    category: TASK_CATEGORY
-    reference: STR_URL | None  # URL to documentation, e.g. published paper
+    type: TASK_TYPE | None = None
+    modalities: list[Literal["text"]] = ["text"]
+    category: TASK_CATEGORY | None = None
+    reference: STR_URL | None = None
 
-    eval_splits: list[str]
+    eval_splits: list[str] = ["test"]
     eval_langs: LANGUAGES
-    main_score: str  # Might want a literal here
+    main_score: str
 
-    date: tuple[STR_DATE, STR_DATE] | None  # When the data was collected
-    domains: list[TASK_DOMAIN] | None
-    task_subtypes: list[TASK_SUBTYPE] | None
-    license: str | None
+    date: tuple[STR_DATE, STR_DATE] | None = None
+    domains: list[TASK_DOMAIN] | None = None
+    task_subtypes: list[TASK_SUBTYPE] | None = None
+    license: str | None = None
 
-    annotations_creators: ANNOTATOR_TYPE | None
-    dialect: list[str] | None
+    annotations_creators: ANNOTATOR_TYPE | None = None
+    dialect: list[str] | None = None
 
-    sample_creation: SAMPLE_CREATION_METHOD | None
-    bibtex_citation: str | None
+    sample_creation: SAMPLE_CREATION_METHOD | None = None
+    bibtex_citation: str | None = None
 
-    descriptive_stats: dict[METRIC_NAME, Optional[dict[SPLIT_NAME, METRIC_VALUE]]]
+    descriptive_stats: dict[METRIC_NAME, Optional[dict[SPLIT_NAME, METRIC_VALUE]]] = {}
+
+    def validate_metadata(self):
+        self.dataset_path_is_specified(self.dataset)
+        self.dataset_revision_is_specified(self.dataset)
+        self.eval_langs_are_valid(self.eval_langs)
+        return True
+
+    @field_validator("dataset")
+    def _check_dataset_path_is_specified(
+        cls, dataset: dict[str, Any]
+    ) -> dict[str, Any]:
+        cls.dataset_path_is_specified(dataset)
+        return dataset
 
     @field_validator("dataset")
-    def _check_dataset_path_is_specified(cls, dataset):
+    def _check_dataset_revision_is_specified(
+        cls, dataset: dict[str, Any]
+    ) -> dict[str, Any]:
+        cls.dataset_revision_is_specified(dataset)
+        return dataset
+
+    @staticmethod
+    def dataset_path_is_specified(dataset: dict[str, Any]) -> bool:
```

Comment:
Here it seems to either raise an error or return a value that will not be used later on. For static methods like this, maybe we don't need a return value. Would suggest removing the return line as well.
```suggestion
    def dataset_path_is_specified(dataset: dict[str, Any]) -> None:
```

Reply:
Agree - I how now reverted it to None